app/api/chat/route.ts

```diff
@@ -0,0 +1,38 @@
+import { Configuration, OpenAIApi } from "openai-edge";
+import { OpenAIStream, StreamingTextResponse } from "ai";
+
+export const runtime = "edge";
+
+const apiConfig = new Configuration({
+  apiKey: "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",
+});
+const openai = new OpenAIApi(apiConfig);
+
+export async function POST(req: Request) {
+  // Extract the `messages` from the body of the request
+  const { messages } = await req.json();
+
+  // Request the OpenAI API for the response based on the prompt
+  const response = await openai.createChatCompletion({
+    model: "gpt-3.5-turbo",
+    stream: true,
+    messages: [
+      { role: "system", content: "You are a helpful assistant." },
```

Comment:
Move to a seperate  configuration 


const message =  {
messageType1: [
  { role: "user", content: "Knock knock." },
      { role: "assistant", content: "Who's there?" },
      { role: "user", content: "Orange." },]
}